### PR TITLE
feat(admin): configure facility

### PR DESCRIPTION
**Pull request by bodnara**
* added option for create & configure when creating facility
* created configuration guide that allows to set:
  * hosts
  * owners
  * preconfigure services
  * required attributes(optional)
  * destinations(optional)
  * managers

### CODE REVIEW

**Review by xkostka2 — CHANGES_REQUESTED** [review]
image
Got this error when getting attributes for service at the "Configure services" tab

Please disable routing in the "Services" at "Summary" tab.

The warning which shows that service destination does not belong to any assigned service is almost never displayed (I managed to trigger it only once by accident)

Would it be possible to at least display some warning (or disable routing to URL with configuration) when the user doesn't come there from facility creation?

**Review by xkureck — CHANGES_REQUESTED** [review]
image
hide the checkboxes in the table they are useless

image
So on this screenshot
1. when I change some attributes but don't save them when I click on next step no dialog appears
2. its kind of weird to show here attributes that you cant change(eg. Max UID) can we do something about it? maybe its a task for backend but I guess in first step we can also hide it on frontend
image
1. apply right color for the checkboxes (in general)
2. specifically to this checkbox - I think that you should reset to default what is checked in this checkbox because its confusing when I check something and then add some service and the when I get back to this page the same set of services are chosen, 


to discuss on the review or daily scrum - picking of the services plus picking services packages is in my opinion weird and maybe should be redesigned+

also consistency - on summary of managers you have title and on the others you don't, I must say that I quite like that the tabs have titles when you open them
image
image

and fix some eslint warnings

File: apps/admin-gui/src/app/facilities/pages/facility-configuration-page/facility-configuration-page.component.html [hunk 1]
```diff
@@ -0,0 +1,232 @@
+<div class="facility-theme">
+  <h1 class="page-title">
```
[the remark this answers is not in the document]
Comment (xkureck) [reply apps/admin-gui/src/app/facilities/pages/facility-configuration-page/facility-configuration-page.component.html:2]: this suggestion

**Review by xkureck — CHANGES_REQUESTED** [review]
image
nothing happens when i click on skip

image
missing subtitle of the page - "Services", also please hide the column "script"

image
missing subtitle of the page "Configure services"

image
On this page wrong sub title and missing info box
image

**Review by xkostka2 — CHANGES_REQUESTED** [review]
image
I managed to open disabled accordion. I entered Summary, with a destination selected, opened accordion, removed destination and the accordion in Summary remained open. 
I suggest closing accordions when leaving Summary. Also, please when accordion is disabled, change the cursor on-hover from pointer to classic cursor.

image
Question: Why isn't there an option to keep the facility as it is in previous tabs? I know there was some reason for it but I can't remember what was it.

**Review by HejdaJakub — CHANGES_REQUESTED** [review]
- When some service is selected, but `NO` service destination is added, then the service destinations are disabled on summary. Other empty parts of summary are enabled to show, so it is little inconsistent.
- Please block `Create & configure` button for `As copy of` other facility. We will probably implement this functionality further, but now this combination throws an error, so we should disable this option in this moment.
- Now is very easy to mistakenly close the configuratoin facility guide without any feedback by clicking to left menu. I am not sure how difficult will be to manage the side menu, but if it is possible, there would be cool to open dialog (the same as is shown after click on cancel button) after click on some item of side menu, when user is still in configuration process.

**Comment by bodnara** [comment]
Fixed accordion behaviour (consistency, cursor). 
Disabled configuration when copying facility. 
Added deactivate guard to prevent accidental configuration exit.

**Comment by HejdaJakub** [comment]
> Question: Why isn't there an option to keep the facility as it is in previous tabs? I know there was some reason for it but I can't remember what was it.
>
It's because you have a `Finish` button on the `Summary` page and if you show button `Keep facility` in this dialog, these buttons (`Finish` and `Keep facility`) would do the same thing, but the different lables can be misleading. If you change lable in the dialog to `Finish`, it can be weird that you have `Finish` button on the `Summary` page and also in the cancel dialog and user could expect different behavior after clicking on these two buttons. But these buttons have the same behavior, so the button in the cancel dialog is duplicate. @xkostka2

**Review by HejdaJakub — CHANGES_REQUESTED** [review]
I found one more broken use case. If I click at step 3 Services on `Next` and `Skip` (no service selected), then click on `Next` from step 6 Managers to step 7 Summary and then click on `Back` to step 6 Managers, it will break the `Next` button (nothing happens when I try to click on it) and also accordion is weird (step 3 Services looks like disabled). Otherwise I think it looks really good.
image

**Review by xkostka2 — CHANGES_REQUESTED** [review]
image

Sorry that I forgot to mention it in the previous review. When there are no hosts or owners, the accordion isn't disabled and when there are no services then the accordion is disabled. Can you unify this, please? Disabling accordions of configurations when there are no services is ok for me.

**Review by xkostka2 — CHANGES_REQUESTED** [review]
Sorry for another thing I missed before. I think we shouldn't allow users to remove themselves from role facility admin when configuring a facility or at least show them some warning that they won't be able to manipulate with facility and terminate the facility configuration.

**Comment by bodnara** [comment]
> Sorry for another thing I missed before. I think we shouldn't allow users to remove themselves from role facility admin when configuring a facility or at least show them some warning that they won't be able to manipulate with facility and terminate the facility configuration.

We moved this step as last to prevend this happening accidentally. However I believe we wanted to allow this if it's intendencional. Mostly for the use case when our support will be configuring facility for someone and would want to remove themself at the end.

**Review by xkostka2 — APPROVED** [review]

**Review by HejdaJakub — APPROVED** [review]

**Pull request merged**